automate_perf.sh

```diff
@@ -1,85 +1,183 @@
+  echo "-t | --capture-to-time        To time for capturing results"
+  echo "h     Print this Help."
+  exit;
+}
+
+SHORT=c,f:,t:,h
+LONG=capture-results-only,capture-from-time:,capture-to-time:,help
+OPTS=$(getopt -a -n automate_perf.sh --options $SHORT --longoptions $LONG -- "$@")
+
+VALID_ARGUMENTS=$# # Returns the count of arguments that are in short or long options
+
+# if [ "$VALID_ARGUMENTS" -eq 0 ]; then
+#   Help
+# fi
+
+eval set -- "$OPTS"
+
+while :
+do
+  case "$1" in
+    -c | --capture-results-only )
+      captureResultsOnly="true"
+      shift 1
+      ;;
+    -f | --capture-from-time )
+      captureFromTime="$2"
+      shift 2
+      ;;
+    -t | --capture-to-time )
+      captureToTime="$2"
+      shift 2
+      ;;
+    -h | --help)
+      Help
+      ;;
+    --)
+      shift;
+      break;
+      ;;
+    *)
+      echo "ERROR: Unexpected option: $1"
+      echo
+      Help
+      ;;
+  esac
+done
+
+if [ "$captureResultsOnly" ] && ([ -z "$captureFromTime" ] || [ -z "$captureToTime" ])
+then
+  echo "ERROR: capture-from-time and capture-to-time should be provided in capture-result-only mode"
+  echo
+  Help
+fi
+
 source ./automate_perf.env
 
 # define the list of dashboards
-declare -a dashboards=("dashboard-account-lookup-service" "Docker%20Prometheus%20Monitoring" "NodeJS%20Application%20Dashboard" "Official%20k6%20Test%20Result")
+declare -a dashboards=(\
+  "dashboard-account-lookup-service" \
+  "Central%20Ledger%20DB" \
+  "Kafka%20Exporter%20Overview" \
+  "mojaloop-central-ledger" \
+  "Mojaloop%20-%20Central-Ledger%20-%20Performance%20Characterization" \
+  "Mojaloop%20-%20ML-API" \
+  "Docker%20Prometheus%20Monitoring" \
+  "NodeJS%20Application%20Dashboard" \
+  "Official%20k6%20Test%20Result" \
+  "MySQL%20Overview" \
+  "Supporting%20Services%20-%20Callback%20Hander%20Service"
+  )
 
-# store current time in a variable
-echo "Start Time : $(date +"%T")"
-startTestSeconds=$(date +"%s")
-# get milliseconds
-startTestMilliseconds=$(date +"%3N")
+# # create a directory to store the results with date timestamp in the name, check if the directory exists
+resultsSubDir="$(date +"%Y%m%d")"
 
+if [ ! -d "results/$resultsSubDir/$K6_SCENARIO_NAME" ]; then
+  mkdir -p results/$resultsSubDir/$K6_SCENARIO_NAME
+fi
 
-env K6_SCRIPT_CONFIG_FILE_NAME=$K6_SCENARIO_CONFIG docker compose --project-name load -f docker-compose-load.yml up -d
+if [ -z "$captureResultsOnly" ]
+then
+  # store current time in a variable
+  echo "Start Time : $(date +"%T")"
+  startTestSeconds=$(date +"%s")
+  # get milliseconds
+  echo "Epoch Start Time : $(date +"%s%3N")"
+  startTestMilliseconds=$(date +"%s%3N")
+
+  if [ ! -d "results/$resultsSubDir/$K6_SCENARIO_NAME/logs" ]; then
+    mkdir -p results/$resultsSubDir/$K6_SCENARIO_NAME/logs
+  fi
+  env K6_SCRIPT_CONFIG_FILE_NAME=$K6_SCENARIO_CONFIG docker compose --project-name load -f docker-compose-load.yml up > results/$resultsSubDir/$K6_SCENARIO_NAME/logs/load-k6-1.log
```

Comment:
Can we change load-k6-1.log to use CONTAINER_NAME variable defined in the script?

Reply:
Addressed by https://github.com/mojaloop/ml-core-test-harness/pull/22/commits/b322162639201131d7ca92c804fe1a79916ef815